fdbserver/ClusterController.actor.cpp

```diff
@@ -3959,7 +3951,15 @@ ACTOR Future<Void> timeKeeper(ClusterControllerData* self) {
 				Version v = tr->getReadVersion().get();
 				int64_t currentTime = (int64_t)now();
 				versionMap.set(tr, currentTime, v);
+				if (!g_network->isSimulated()) {
```

Comment:
One consequence of moving this block is that if you have disabled timekeeper, it will no longer run the debug transaction. We can avoid that by reordering this code to do the following:

1. wait for disable key
2. Get current time and version
3. Run the debug transaction
4. Check the disable value and break if necessary
5. Update the version map

Reply:
Oh I think I was confused, this block doesn't run the debug transaction but just enables it. It's fine as is.

Reply:
Thinking about this more carefully, it doesn't have the original problem I described but I think it does change the behavior.

In this version, we won't have enabled the debug transaction when we do our first read, so we won't have a timing of the GRV call (or the read, which is perhaps not as interesting). So maybe we should leave the enabling where it was and only move the trace event?

Reply:
Good point! Fix applied.